homeworks/08-functional-reactive-programming/src/game/gameplay.ts

```diff
@@ -0,0 +1,25 @@
+import { from, interval, Subject } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { CellContent } from '../types';
+import { generateGrid, getRandomTarget } from '../utils';
+
+const shotTarget = new Subject<CellContent>();
+export const shotTarget$ = shotTarget.asObservable();
+
+const targetIndex$ = interval(1000).pipe(map(() => getRandomTarget()));
+
+const emptyGrid = generateGrid();
+
+export const gameGrid$ = from(targetIndex$)
+  .pipe(
+    map((targetIndex) => {
+      const newGrid = JSON.parse(JSON.stringify(emptyGrid));
```

Comment:
It's better to use spread operator to copy an array instead of `JSON.parse JSON.stringify`. This works faster and preserves typings. However spread creates just a shallow copy of an array you need to change the code like that to keep all grid items as they are:
```typescript
const newGrid = [...emptyGrid];
newGrid[targetIndex] = { ...newGrid[targetIndex], content: CellContent.CAT };
return newGrid;
```

Reply:
Fixed in https://github.com/vbarbasins/evo-ts-bootcamp/pull/9/commits/4c0a413a4fb45fa92b571474bb4276beb79ec100